StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/VerticalSavedPaymentMethodsViewController.swift

```diff
@@ -109,6 +139,32 @@ class VerticalSavedPaymentMethodsViewController: UIViewController {
     @objc func didSelectEditSavedPaymentMethodsButton() {
         isEditingPaymentMethods = !isEditingPaymentMethods
     }
+
+    private func remove(paymentMethod: STPPaymentMethod) {
+        guard let button = paymentMethodRows.first(where: { $0.paymentMethod.stripeId == paymentMethod.stripeId }),
+                let ephemeralKeySecret = configuration.customer?.ephemeralKeySecret else { return }
+
+        // Detach the payment method from the customer
+        let manager = SavedPaymentMethodManager(configuration: configuration)
+        manager.detach(paymentMethod: paymentMethod, using: ephemeralKeySecret)
+
+        // Remove the payment method row button
+        paymentMethodRows.removeAll { $0.paymentMethod.stripeId == paymentMethod.stripeId }
+        stackView.removeArrangedSubview(button, animated: true)
+
+        // Update the editing state if needed
+        isEditingPaymentMethods = canEdit
+
+        // If we deleted the last payment method kick back out to the main screen
+        if paymentMethodRows.isEmpty {
+            completeSelection()
+        }
+    }
+
+    private func completeSelection() {
+        _ = self.bottomSheetController?.popContentViewController()
+        self.delegate?.didComplete(with: selectedPaymentMethod, latestPaymentMethods: paymentMethods)
```

Comment:
General best practice is to let the thing that presented/pushed you also pop/dismiss you. In this case I think we'll want the delegate to control when to pop the sheet so that it can re-render the list of payment methods (possibly removing the saved PM row) before popping.